core/storage/amt/amt.cpp

```diff
@@ -369,4 +371,19 @@ namespace fc::storage::amt {
   uint64_t Amt::maxAt(uint64_t height) const {
     return maskAt(height + 1);
   }
+
+  void Amt::lazyCreateRoot() const {
+    if (which<std::monostate>(root_)) {
+      Root root;
+      if (v3()) {
+        root.bits = bits();
+      }
+      root.node.bits_bytes = bitsBytes();
+      root_ = std::move(root);
+    }
+  }
+
+  bool Amt::v3() const {
+    return ipld->actor_version >= vm::actor::ActorVersion::kVersion3;
+  }
```

Comment:
Why do we need it?
All version have bitwidth, v0 & v2 just have default value.
We can use `bits` for all versions without specific behaviour for `v3()` cases, no?

Reply:
We discussed, that `Array`/`Map` should rely on actor_version like `Universal`.
To avoid unnecessary duplication like `InitActorState { Map<v3=false> map0; Map<v3=true> map3; }`.